src/discordtipbot/tipbot.cr

```diff
@@ -265,6 +265,6 @@ class TipBot
   end
 
   private def balance(id : UInt64)
-    @db.query_one("SELECT balance FROM accounts WHERE userid=$1", id, &.read(Float64)) || 0.0
+    @db.query_one("SELECT balance FROM accounts WHERE userid=$1", id, &.read(BigDecimal)) || BigDecimal.new("0")
```

Comment:
(ditto my other comment about the correct overload usage here)

But, this is also the wrong method, as this will raise `DB::Error` instead of returning `nil`. I believe here you want to use `query_one?`.
<http://crystal-lang.github.io/crystal-db/api/0.5.0/DB/QueryMethods.html#query_one%3F%28query%2C%2Aargs%2Castype%3AClass%29-instance-method>

However, that doesn't really make sense to return a balance of 0 when there is no account (it's technically wrong).